File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        for(int i = 0;i<nums.length-2;i++) {
        	for(int j = i+1;j<nums.length-1;j++) {
        		for(int k = j+1;k<nums.length;k++) {
        			int p;
        			if(Math.sqrt(nums[i]+nums[j]+nums[k])==(int)Math.sqrt(nums[i]+nums[j]+nums[k]))
        				continue;
        			for(p = 2;p<Math.sqrt(nums[i]+nums[j]+nums[k]);p++) {
        				if((nums[i]+nums[j]+nums[k])%p==0)
        					break;
        			}
        			if(p==(int)Math.sqrt(nums[i]+nums[j]+nums[k])+1) {
        				answer++;
        			}
        			
        		}
        	}
        }

        return answer;
    }
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.cpp
#include <vector>
#include <iostream>
using namespace std;
bool isPrime(int num) {
    for (int i = 2;i * i <= num;i++) {
        if (!(num % i))
            return false;
    }
    return true;
}

int solution(vector<int> nums) {
    int answer = 0;
    
    for (int i = 0;i < nums.size()-2; i++) {
        for (int j = i + 1;j < nums.size() - 1;j++) {
            for (int k = j + 1;k < nums.size(); k++) {
                if (isPrime(nums[i] + nums[j] + nums[k]))
                    answer++;
            }
        }
    }
    return answer;
}